controllers/quay/quayregistry_controller.go

```diff
@@ -84,6 +87,22 @@ func (r *QuayRegistryReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	updatedQuay := quay.DeepCopy()
 	quayContext := quaycontext.NewQuayRegistryContext()
 
+	isQuayMarkedToBeDeleted := quay.GetDeletionTimestamp() != nil
+	if isQuayMarkedToBeDeleted {
+		if controllerutil.ContainsFinalizer(updatedQuay, QuayOperatorFinalizer) {
+			if err := r.finalizeQuay(ctx, updatedQuay); err != nil {
+				return ctrl.Result{}, err
```

Comment:
In the future, if the finalizer fails we might want to emit an event and set  a `status.condition` to convey the error to the user. But this is fine for now.